abel/basex.py

```diff
@@ -163,22 +257,59 @@ def _get_left_right_matrices(M, Mc):
     M_right = M.dot(inv((M.T.dot(M) + E)))
     return M_left, M_right
 
+def _get_left_right_matrices_asym(M_vert, M_horz, Mc_vert, Mc_horz): 
+    """ An internal helper function for no-up/down-asymmetry BASEX:
+        given basis sets  M_vert, M_horz, Mc_vert, Mc_horz,
+        return M_left and M_right matrices 
+    """
+
+    nbf_vert, nbf_horz = np.shape(M_vert)[1], np.shape(M_horz)[1]
+    
+    q_vert, q_horz = 0,0 # No Tikhonov regularization
+    E_vert, E_horz = np.identity(nbf_vert)*q_vert, np.identity(nbf_horz)*q_horz
+
+    vert_left = inv(Mc_vert.T.dot(Mc_vert) + E_vert).dot(Mc_vert.T)
+    horz_right = M_horz.dot(inv(M_horz.T.dot(M_horz) + E_horz))
+
+    return vert_left, horz_right
 
 def _nbf_default(n, nbf):
-    """ An internal helper function to check that nbf = n//2 and print a waring
+    """ An internal helper function to check that nbf = n//2 + 1 and print a warning
     otherwise """
     if nbf == 'auto':
         nbf = n//2 + 1
     else:
         if nbf != n//2 +1:
-            print('Warning: the number of basis functions nbf = {} != (n-1)/2 = {}\n'.format(n, nbf),
+            print('Warning: the number of basis functions nbf = {} != (n//2 + 1) = {}\n'.format(n, nbf),
                     '    This behaviour is currently not tested and should not be used\
                     unless you know exactly what you are doing. Setting nbf="auto" is best for now.')
 
     return nbf
 
+def _nbf_default_asym(n_vert, n_horz, nbf):
+    """ An internal helper function for the asymmetric case to check that nbf = n//2 + 1 and print a warning otherwise """
+    if nbf == 'auto':
+        # nbf_vert = n_vert (if relevant)
+        # nbf_horz = n_horz//2 + 1
+        nbf = [n_vert, n_horz//2 + 1]
+    elif type(nbf) == int:
+        if nbf != n_horz//2 +1:
+            print('Warning: the number of basis functions nbf = {} != (n//2 + 1)  = {}\n'.format(nbf, n_horz//2 +1),
+                    '    This behaviour is currently not tested and should not be used\
+                    unless you know exactly what you are doing. Setting nbf="auto" is best for now.')
+        nbf = [nbf]*2 # Setting identical number of vert and horz functions
+    elif type(nbf) == list:
```

Comment:
also in the case nbf is a tuple, maybe rather do something like `elif isinstance(nbf, (list, tuple)):`